controllers/osbuildenvconfig_controller.go

```diff
@@ -278,6 +344,106 @@ func (r *OSBuildEnvConfigReconciler) Update(ctx context.Context, reqLogger logr.
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	created, err = r.ensureWorkerConfigAnsibleConfigExists(ctx, instance)
```

Comment:
The update function becomes to be a too long function that includes too many update cases
I think it's better to export each update to a separate function - first part for ensure composer resources and the second part will be for the workers